client/my-sites/plans-features-main/index.tsx

```diff
@@ -291,32 +292,44 @@ const PlansFeaturesMain = ( {
 	const intentFromSiteMeta = usePlanIntentFromSiteMeta();
 	const planFromUpsells = usePlanFromUpsells();
 	const [ forceDefaultPlans, setForceDefaultPlans ] = useState( false );
-
 	const [ intent, setIntent ] = useState< PlansIntent | undefined >( undefined );
+
+	const { segment: intentFromSegmentationSurvey, isLoadingSegment } = useSegmentedIntent(
+		flowName === 'guided',
+		siteId
+	);
```

Comment:
I may be missing something but is there anything blocking from moving this call to the consumer of `plans-features-main` and avoid the unnecessary logic/conditioning in the main hook below? Just pass the intent you want into `plans-features-main`. That's what the `intent` prop is meant for.

You can still call it here to get the `isLoadiingSegment` state if that's meaningful to anything, but I'd rather we avoid `intentFromSegmentationSurvey`. You are overriding the `intentFromProps` indirectly with this. I would not expect any changes to the `useEffect` hook below 🤷‍♂️

Reply:
That's when I started, really. But the parent component is a class component, and doing network requests there sucks. I'd have to make Redux actions, selector, reducer, etc... I can do all of these, but IMHO the trade-off is not worth it. 4 lines vs a bunch of files.

> You are overriding the intentFromProps indirectly with this. 

That's a great point! I wasn't worried about it because the hook only engages for this flow. But it's still code smell to ignore props.

How about [this](https://github.com/Automattic/wp-calypso/pull/90809/commits/d9636d4d6eae209764884de091e69114c4bde322)?

Reply:
Hey Omar. I haven't had a chance to check the last update. Concerning the class component though, you can write a simple HOC in that case, call the hook and propagate the value in the props. It can all be done in the same file, 3-4 lines wrapping the connected component, no?

Reply:
^ that's been our strategy for cases in plans-grid that we needed the interplay of hooks and classes, before refactoring everything to function form